src/Protocol/Protocol_1_11.cpp

```diff
@@ -388,7 +388,7 @@ void cProtocol_1_11_0::SendSpawnMob(const cMonster & a_Mob)
 	// TODO: Bad way to write a UUID, and it's not a true UUID, but this is functional for now.
 	Pkt.WriteBEUInt64(0);
 	Pkt.WriteBEUInt64(a_Mob.GetUniqueID());
-	Pkt.WriteVarInt32(static_cast<UInt32>(a_Mob.GetMobType()));
+	Pkt.WriteVarInt32(static_cast<UInt32>(GetProtocolMobID(a_Mob.GetMobType())));
```

Comment:
Please please please don't use `ID` for things that are not IDs. It's a Type, not an ID.
(ID = unique identifier, unique for each instance. Type = kind of an object, possibly shared by multiple objects)

Reply:
Also, you should first check if `GetProtocolMobID` returns a valid type, and only then spawn the mob (what happens to the client, if you send a packet with an invalid mob type? What happens if you don't spawn a mob but then send movement and action packets to it?)

Reply:
Sigh, tell that to Mojang...

Reply:
I think we should just remap newer mobs to some other mob in older MC versions. If not, we would get an "invisible" mob in any case, which we don't want more of.